File: _site/python_solutions/342.power-of-four.py
#
# @lc app=leetcode id=342 lang=python
#
# [342] Power of Four
#
# https://leetcode.com/problems/power-of-four/description/
#
# algorithms
# Easy (39.80%)
# Total Accepted:    101.4K
# Total Submissions: 254.7K
# Testcase Example:  '16'
#
# Given an integer (signed 32 bits), write a function to check whether it is a
# power of 4.
# 
# Example 1:
# 
# 
# Input: 16
# Output: true
# 
# 
# 
# Example 2:
# 
# 
# Input: 5
# Output: false
# 
# 
# Follow up: Could you solve it without loops/recursion?
#
class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        return num & (num - 1) == 0 and (num - 1) % 3 == 0

File: _site/solutions/342_power_of_four.rb
# Given an integer (signed 32 bits), write a function to check whether it is a power of 4.
#
# Example:
# Given num = 16, return true. Given num = 5, return false.
#
# Follow up: Could you solve it without loops/recursion?

# @param {Integer} num
# @return {Boolean}
def is_power_of_four(num)
  num & (num - 1) == 0 && (num - 1) % 3 == 0
end

(0..10_000).each do |i|
  p i if is_power_of_four(i)
end
